src/node_crypto.cc

```diff
@@ -4846,6 +4849,85 @@ static int GetDefaultSignPadding(const ManagedEVPPKey& key) {
                                                       RSA_PKCS1_PADDING;
 }
 
+static const unsigned int kNoDsaSignature = static_cast<unsigned int>(-1);
+
+static unsigned int GetBytesOfRS(const ManagedEVPPKey& pkey) {
+  int bits, base_id = EVP_PKEY_base_id(pkey.get());
+
+  if (base_id == EVP_PKEY_DSA) {
+    DSA* dsa_key = EVP_PKEY_get0_DSA(pkey.get());
+    bits = BN_num_bits(DSA_get0_q(dsa_key));
```

Comment:
You're using `q` because `q < p`, right? Maybe a comment is in order because I suspect that isn't very obvious to the casual reader.

Reply:
Mostly because `r` and `s` are computed `mod q`. Added a comment.